### ISSUE TITLE
Imports are added in between multiline package statements

### ISSUE THREAD

**Issue by matthughes** [scalameta/metals#1709]
**Describe the bug**
Given a file with multiline package statements:

```scala
package a
package b
package c
```
when Metals in VsCode goes to add an import, it always adds it after the first package statement, instead of at the end of all package statements.

**To Reproduce**
1. Bring up a scala file with multiple package statements
2. Attempt to import a symbol
3. See that symbol was imported in between package statements

**Expected behavior**
Imports should be added after all package statements

**Installation:**
- Operating system: macOS/
- Editor: Visual Studio Code/
- Metals version: v0.9.0

**Comment by tgodzik** [comment]
Thanks for reporting! This is most likely due to https://github.com/scalameta/metals/pull/1686 

We need to figure out if we can detect the case better.

**Comment by colineto** [comment]
Ah interesting ! yes sure I'll take a look

**Comment by tgodzik** [comment]
Thanks!

**Comment by colineto** [comment]
Hey @tgodzik This is definitely due to #1686. Which means the remaining issue is with `package object` and I think the only good solution is to have a distinction between a `package` and a `package object`

So I can do 
```scala
val enclosingPackage = lastVisitedParentTrees.collectFirst {
  case pkg: PackageDef 
    if(!pkg.children.exists(_.symbol.toString.contains("package object"))) => 
      pkg
}
```
which works well

But I'm wondering how much we should modify this to take `package object` into account
```scala
abstract class MemberDef extends DefTree with MemberDefApi {
    def mods: Modifiers
    def keyword: String = this match {
      case TypeDef(_, _, _, _)      => "type"
      case ClassDef(mods, _, _, _)  => if (mods hasFlag TRAIT) "trait" else "class"
      case DefDef(_, _, _, _, _, _) => "def"
      case ModuleDef(_, _, _)       => "object"
      case PackageDef(_, _)         => "package"
      case ValDef(mods, _, _, _)    => if (mods hasFlag MUTABLE) "var" else "val"
      case _ => ""
    }
  }
```

**Comment by tgodzik** [comment]
@colineto Do you mean fixing it in the compiler? That for sure would be useful, however we can't really do it for any current Scala versions, so we still need to do it in Metals. 

I would modify your check to maybe something like:
```scala
        def notPackageObject(pkg: PackageDef) = pkg.stats.exists {
          case ModuleDef(_, name, _) => name.toString != "package"
          case _ => true
        }
```

We make sure here that it is not an object named package, which I think is safer a bit.  Might not be needed.

**Comment by tgodzik** [comment]
Closed by #1722
